### ISSUE TITLE
Viewer runs twice

### ISSUE THREAD

**Issue by MaxwellSalmon** [lfoppiano/streamlit-pdf-viewer#70]
Thanks for making such a great component, it's very nice!

However, I am experiencing issues, especially when viewing large PDF files, where it seems the viewer reloads once the PDF file has loaded, making the code run twice.

I tried making a simple page in my dashboard to test it. Here's the code:

```
from streamlit_pdf_viewer import pdf_viewer
import streamlit as st
import time

print(time.time())
pdf_viewer(st.session_state['pdf'], height=950,
            width=764,
            scroll_to_annotation=1,
            annotations=[st.session_state['annotations']],
            )
print("test")
```

This will print the time and "test", then the PDF is rendered. Once it is completely loaded, the statements are printed to the console again. This gets much more prominent, when `scroll_to_annotation` is uncluded, as it will make the whole page jump around, especially with long PDF's.

Do you have any suggestions to why this is happening or any ideas to how I can work around this?

Thanks!

**Comment by MaxwellSalmon** [comment]
Thanks for the quick response! 

It's actually an issue with with both long and short documents, but most disturbing with long documents, as it takes loner to load both times.

 The one I tested was +70 pages, but I have now tested with a 9-page long document, which have the same issue. The issue is also more prominent, on my main page, which features other functionalities, such as input fields.

**Comment by lfoppiano** [comment]
OK. Which browser and with Operating System do you use?

**Comment by lfoppiano** [comment]
@MaxwellSalmon I'm sorry, I haven't used it with Edge, as I don't have Windows, I cannot guarantee support for Edge. My time is limited and I wish I have a contributor helping improving the support for Windows/Edge. 

Could you give it a try with Firefox, Chrome or Brave (nicer version of Chrome) and let me know whether it's working better?

**Comment by MaxwellSalmon** [comment]
No worries, I completely understand! I have tried it with Chrome, which gave the same issue, though I am unable to install other browsers on this PC at the moment.

**Comment by lfoppiano** [comment]
OK, can you share a couple of PDF documents? You can send them to luca AT Sciencialab.com if they cannot be shared online.

**Comment by MaxwellSalmon** [comment]
I am afraid, I can't. The documents I am working with are medical documents with sensitive information. Do you suspect the PDF is to blame?

**Comment by lfoppiano** [comment]
Hi @MaxwellSalmon no problem, I understand. I did tried with some documents I have at hand but for me it works fine. Without reproducing the issue locally I'm not sure how to solve it 😄 

@t29mato when you have some time, could you please go through the JS part and double-check that we don't refresh the page twice?

**Comment by lfoppiano** [comment]
Hi @MaxwellSalmon, could you please try version 0.0.19-dev0? I think your concern about the double rendering was concrete, and I should have fixed it, hopefully. 

At least the rendering should have been improved, but I'm still working on the page transition, which should happens only after the whole DOM is loaded, however there might still be room for improvement.

**Comment by MaxwellSalmon** [comment]
Hello @lfoppiano 

Thanks a lot for the update! I tried it out and it indeed seems faster. Yet, the program above still runs twice. Here's the output:

```
1729238847.6578197
test
1729238856.4343493
test
```

The PDF file has 37 pages. If you have any ideas, it will be much appreciated, otherwise we have decided to only present a part of the PDF to the user 🙂

**Comment by t29mato** [comment]
@lfoppiano 
The twice execute comes from the following line:

https://github.com/lfoppiano/streamlit-pdf-viewer/blob/v0.0.19-dev1/streamlit_pdf_viewer/frontend/src/PdfViewer.vue#L307

```javascript
      Streamlit.setComponentValue({"pages": Array.from(pages_ids), "annotations": Array.from(annotations_ids)})
```


> When you call Streamlit.setComponentValue(new_value), that new value is sent to Streamlit, which then re-executes the Python script from top to bottom. When the script is re-executed, the call to my_component(...) will return the new value.

As mentioned in the [Streamlit documentation](https://docs.streamlit.io/develop/concepts/custom-components/intro) above, the method re-executes the Python script.

Then, I couldn't find the code to use those values `pages` and `annotations` in Python.

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by MaxwellSalmon** [comment]
This solved the issue! Thank you so much 😄

**Comment by lfoppiano** [comment]
@MaxwellSalmon thanks! I've released version 0.0.19
